purchase_work_acceptance_late_fine/models/work_acceptance.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2020 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class WorkAcceptance(models.Model):
+    _inherit = "work.acceptance"
+    _description = "Work Acceptance"
+
+    late_days = fields.Integer(string="Late Days", compute="_compute_late_days",)
+    fine_rate = fields.Monetary(
+        string="Fine Rate", default=lambda self: self.env.company.wa_fine_rate,
+    )
+    late_fine = fields.Monetary(
+        string="Fine Amount", compute="_compute_late_fine", store=True,
+    )
+
+    def _prepare_account_move_line(self, move):
+        return {
+            "name": "%s" % (self.name),
+            "move_id": move.id,
+            "date_maturity": move.invoice_date_due,
+            "partner_id": move.partner_id.id,
+        }
+
+    @api.depends("date_accept", "date_due")
+    def _compute_late_days(self):
```

Comment:
Compute field is multi by nature.
Only if date_accept and date_due has value, we will continue calculate., i.e.,
```
for rec in self:
  if not rec.date_accept or not rec.date_due:
    continue
  rec.late_days = (rec.date_accept - rec.date_due).days
```

Reply:
Note: your logic is quite confusing. I am not sure why you try to loop through date? Or the variable naming is not ok.